README.md

```diff
@@ -383,6 +429,72 @@ combined
     └── test_candidates_deezymatch.pkl
 ```
 
+In case `query_candidate_dirname` was set in [changing query/candidate directory names](#changing-querycandidate-directory-names), the vector representations can be combined by:
+
+```python
+from DeezyMatch import combine_vecs
+
+# combine vectors
+combine_vecs(qc_modes='q', 
+             rnn_passes=['fwd', 'bwd'], 
+             input_scenario='test', 
+             query_candidate_dirname='my_query_dir',
+             output_scenario='test', 
+             print_every=10)
+```
+
+(and similarly for candidate vectors).
+
+Moreover, it is possible to change the default dirname where combined vector representations are stored (by default, it is `combined`, see the above directory structure):
+
+```python
+from DeezyMatch import combine_vecs
+
+# combine vectors
+combine_vecs(qc_modes='q', 
+             rnn_passes=['fwd', 'bwd'], 
+             input_scenario='test', 
+             output_scenario='test', 
+             query_candidate_dirname='my_query_dir',
+             output_par_dir="my_combined_dir",
+             print_every=10)
+```
+
+:warning: in this case, `output_par_dir="my_combined_dir"` should be set for `qc_modes='c'` as well, that is:
+
+```python
+from DeezyMatch import combine_vecs
+
+# combine vectors
+combine_vecs(qc_modes='c', 
+             rnn_passes=['fwd', 'bwd'], 
+             input_scenario='test', 
+             output_scenario='test', 
+             query_candidate_dirname='my_candidate_dir',
+             output_par_dir="my_combined_dir",
+             print_every=10)
+```
+
+which results in the following directory structure:
+
+```
+my_combined_dir/
+└── test
+    ├── candidates_bwd_id.pt
+    ├── candidates_bwd_items.npy
+    ├── candidates_bwd.pt
+    ├── candidates_fwd_id.pt
+    ├── candidates_fwd_items.npy
+    ├── candidates_fwd.pt
+    ├── input_dfm.yaml
+    ├── queries_bwd_id.pt
+    ├── queries_bwd_items.npy
+    ├── queries_bwd.pt
+    ├── queries_fwd_id.pt
+    ├── queries_fwd_items.npy
+    └── queries_fwd.pt
+```
+
```

Comment:
I am wondering whether this could be simplified. I'm afraid it can be a bit confusing for the user.

I think one problem is that we assume that the scenario name will be the same for both `candidates` and `queries`. However, we generate vectors for candidates only once per gazetteer, while we can generate vectors for queries for multiple different real-case datasets per gazetteer (for example, we generate candidate vectors for the `wikigaz_en` gazetteer only once, we generate query vectors for the `WOTR` and `BNA-FMP` datasets that we'll try to match to the `wikigaz_en` gazetteer). So we ideally would have something like `candidates/modelname_gazetteer` and `queries/modelname_wotr` and `queries/modelname_bnafmp`.

If I'm not wrong, the `qc_modes` parameter is used only to specify the directory of the vectors we want to combine, either `candidates` or `queries`. So, if we just specify the input folder, it does not really matter whether we have `q` or `c`, am I right?

This is one suggestion, not sure if the best:
* we remove the `qc_modes` option.
* the user **must** specify the `input_scenario_folder`. This is the full path, can be a string (e.g. `"candidates/modelname_gazetteer/"`) or a list of two strings (e.g. `["candidates/modelname_gazetteer/", "queries/modelname_wotr"]`). This means we'll always have to specify the parent folder if we have the default `candidates/` and `queries/` folders, but also gives more flexibility.
* the user **must** specify the `output_scenario` name.
* the user **can** specify the `output_par_dir`, which by default is `combined/`

For example:
```
combine_vecs(rnn_passes=['fwd', 'bwd'], 
             input_scenario='["candidates/modelname_wikigaz_en/", "queries/modelname_wotr"]', 
             output_scenario='wikigaz_en_wotr',
             print_every=10)
```

In this example, we would combine the vectors in `candidates/modelname_wikigaz_en/` and `queries/modelname_wotr/`, and the output would be stored in `combined/wikigaz_en_wotr` (default value in `output_par_dir` + `output_scenario`).

Sorry for the super long comment. Let me know if you think this does not make sense or makes it more complex. We can have a chat if you want.

Reply:
Hopefully, the commit https://github.com/Living-with-machines/DeezyMatch/commit/4a45874d18da3bb6d878bc6dfc331b9fd7be038c addresses this issue. Please let me know if I missed anything.